src/executables/master.c

```diff
@@ -652,13 +652,14 @@ static void
 css_send_to_existing_server (CSS_CONN_ENTRY * conn, unsigned short rid, CSS_SERVER_REQUEST request)
 {
   SOCKET_QUEUE_ENTRY *temp;
-  char *server_name = NULL;
   int name_length, buffer;
   name_length = 1024;
-  if (css_receive_data (conn, rid, &server_name, &name_length, -1) == NO_ERRORS && server_name != NULL)
+  std::string server_name;
+  SERVER_TYPE type;
+
+  if (receive_server_info (conn, rid, server_name, type) == NO_ERRORS && !server_name.empty ())
```

Comment:
It is best to separate functions that modify one of the args and then using the same arg again in different conditionals. It is easier to read.
```suggestion

  if (receive_server_info (conn, rid, server_name, type) != NO_ERRORS)
    {
      css_free_conn (conn);
      return;
    }

  if (!server_name.empty ())
```

Reply:
also changed to `server_name.empty () == false` for better visibility